src/OlStyler.js

```diff
@@ -182,34 +183,140 @@ function pointStyle(pointsymbolizer) {
+        ? parseText[keys[0]](part[keys[0]], fature)
+        : '');
+    }, '');
+
+    const fill = textsymbolizer.fill ? (textsymbolizer.fill.css || textsymbolizer.fill.svg) : {};
+    const halo = textsymbolizer.halo && textsymbolizer.halo.fill
+      ? (textsymbolizer.halo.fill.css || textsymbolizer.halo.fill.svg)
+      : {};
+    const haloRadius = textsymbolizer.halo && textsymbolizer.halo.radius
+      ? parseFloat(textsymbolizer.halo.radius)
+      : 1;
+    const {
+      fontFamily = 'sans-serif',
+      fontSize = 10,
+      fontStyle = '',
+      fontWeight = '',
+    } = textsymbolizer.font && textsymbolizer.font.css ? textsymbolizer.font.css : {};
+
+    const pointplacement = textsymbolizer && textsymbolizer.labelplacement
+      && textsymbolizer.labelplacement.pointplacement
+      ? textsymbolizer.labelplacement.pointplacement
+      : {};
+    const displacement = pointplacement && pointplacement.displacement
+      ? pointplacement.displacement
+      : {};
+    const offsetX = displacement.displacementx
+      ? displacement.displacementx
+      : 0;
+    const offsetY = displacement.displacementy
+      ? displacement.displacementy
+      : 0;
+    const lineplacement = textsymbolizer && textsymbolizer.labelplacement
+      && textsymbolizer.labelplacement.lineplacement
+      ? textsymbolizer.labelplacement.lineplacement
+      : null;
+    const rotation = pointplacement.rotation
+      ? pointplacement.rotation
+      : 0;
+
+    const placement = type !== 'point' && lineplacement ? 'line' : 'point';
+
+    return new Style({
+      text: new Text({
+        text,
+        font: `${fontStyle} ${fontWeight} ${fontSize}px ${fontFamily}`,
+        offsetX,
+        offsetY,
+        rotation,
+        placement,
+        textAlign: 'center',
+        textBaseline: 'middle',
+        stroke: new Stroke({
+          color: halo.fillOpacity && halo.fill && halo.fill.slice(0, 1) === '#'
+            ? hexToRGB(halo.fill, halo.fillOpacity)
+            : halo.fill,
+          // wrong position width radius equal to 2 or 4
+          width: (haloRadius === 2 || haloRadius === 4)
+            ? haloRadius - 0.00001
+            : haloRadius,
+        }),
+        fill: new Fill({
+          color: fill.fillOpacity && fill.fill && fill.fill.slice(0, 1) === '#'
+            ? hexToRGB(fill.fill, fill.fillOpacity)
+            : fill.fill,
+        }),
+      }),
+    });
+  }
+  return new Style({});
+}
+
 /**
  * Create openlayers style
  * @example OlStyler(getGeometryStyles(rules), geojson.geometry.type);
  * @param {GeometryStyles} GeometryStyles rulesconverter
  * @param {string} type geometry type, @see {@link http://geojson.org|geojson}
  * @return ol.style.Style or array of it
  */
-export default function OlStyler(GeometryStyles, type = 'Polygon') {
-  const { polygon, line, point } = GeometryStyles;
+export default function OlStyler(GeometryStyles, type = 'Polygon', properties) {
```

Comment:
I want to suggest the following:
```suggestion
export default function OlStyler(GeometryStyles, feature) {
const { properties, geometry } = feature;
const { type } = geometry.type;
```
This breaks the api (it will be a new release anyway) but seems more clean to me.

Reply:
Thanks! Yes, I agree with these changes, this looks cleaner.
This will also add the possibility to access coordinates for geometry transformations.
I would like to commit directly from your changes but I think the row 287.
```
const { type } = geometry.type;
```
needs to be changed in 
```
const { type } = geometry;
```

Reply:
Correct :) I edited it

Reply:
Thanks :) !